drivers/bridge/port_mapping.go

```diff
@@ -50,6 +51,12 @@ func (n *bridgeNetwork) allocatePortsInternal(bindings []types.PortBinding, cont
 			bs = append(bs, bIPv4)
 		}
 
+		// skip adding v6 addr when the kernel was booted with `ipv6.disable=1`
+		// https://github.com/moby/moby/issues/42288
+		if !IsV6Listenable() {
```

Comment:
The only case we need to fix here (I think...) is when 0.0.0.0 is passed.
If someone specifies a v6 address and we can't do that I don't think we should ignore it.

Reply:
fixed

Reply:
I don't understand what was fixed.
I think the check for `IsV6Listenable()` (or whwatever its called now) should go into https://github.com/moby/libnetwork/pull/2635/files#diff-5e6a1e1b6a7570d4362c2cfd293b0f05b409b6c9b1c8157b9832f427f6e0cf56L109 explicitly where we are checking if the bind address is `0.0.0.0`

Reply:
`isV6Binding := c.HostIP != nil && c.HostIP.To4() == nil` is set to false when the bind address is `0.0.0.0`, and set to true when the bind address is set to IPv6 explicitly, so the issue was fixed IIUC.

Test matrix:
- `docker run -p 80:80`: listens on 0.0.0.0:80, without an error
- `docker run -p 0.0.0.0:80:80`: listens on 0.0.0.0:80, without an error
- `docker run -p [::]:80:80`:  raises an error

Reply:
My point is we are already checking for 0.0.0.0 when validating the bind address, why not add the check when we do that?

Reply:
It's also a bit less confusing because it is checking `net.IPv4zero` specifically.

Reply:
I tried that, but couldn't make it work

Reply:
Sorry to be a stickler but I don't understand the check you have in place and how it works.
Meanwhile: https://play.golang.org/p/jtQGBQ4yzWd

Reply:
It should be able to `implicitBinding := hostIP != nil && hostIp.Equal(net.IPv4zero)`

Reply:
@tianon helped me unbreak my brain here.
This seems like it's ok because we are checking if it is a v4 binding and only continuing if it is a v6-only binding which makes sense. It may also be that we actually have a nil host IP...